x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlConfiguration.java

```diff
@@ -81,20 +97,52 @@ public Locale locale() {
         return locale;
     }
 
+    public String query() {
+        return query;
+    }
+
+    private static void writeQuery(StreamOutput out, String query) throws IOException {
+        if (query.length() > QUERY_COMPRESS_THRASHOLD_CHARS) { // compare on chars to avoid UTF-8 encoding unless actually required
+            out.writeBoolean(true);
+            var baos = new ByteArrayOutputStream();
+            try (var gzip = new GZIPOutputStream(baos)) {
+                gzip.write(query.getBytes(StandardCharsets.UTF_8));
+            }
+            out.writeByteArray(baos.toByteArray());
+        } else {
+            out.writeBoolean(false);
+            out.writeString(query);
+        }
+    }
+
+    private static String readQuery(StreamInput in) throws IOException {
+        boolean compressed = in.readBoolean();
+        if (compressed) {
+            byte[] bytes = in.readByteArray();
+            var bais = new ByteArrayInputStream(bytes);
+            try (var gzip = new GZIPInputStream(bais)) {
+                return new String(gzip.readAllBytes(), StandardCharsets.UTF_8);
+            }
+        } else {
+            return in.readString();
+        }
+    }
+
```

Comment:
Have you checked `org.elasticsearch.common.compress`?
It relies only on the Deflate/Inflate compressor instead of the whole gzip - in our case since we're doing string compression, we don't expect things to be too large anyway.

Reply:
I have, but seemed heavier than the need here. I switched over though. Thanks.